packages/components/src/Gesture/withListGesture.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import keycode from 'keycode';
+import omit from 'lodash/omit';
+
+function focusOn(event, element) {
+	if (element) {
+		element.focus();
+	}
+}
+
+function getAllItems(ref) {
+	return ref.closest('[role="list"]').querySelectorAll('[role="listitem"]');
+}
+
+function getNextItem(ref) {
```

Comment:
What if just use `element.nextElementSibling` and `element.previousElementSibling` ?